programs/github

```diff
@@ -108,8 +108,11 @@ if action == subcommands.clone then
 	local tree = repo:tree(treeName)
 
 	-- download the files
-	print("Downloading:")
 	local totalSize = tree.size
+	if totalSize > fs.getFreeSpace(dest) then
+		return printError("Repository is to large for download!")
```

Comment:
I guess I don't understand whats happening here... what does the `with` do I'm not familiar with the syntax here...